src/libsyntax/feature_gate/check.rs

```diff
@@ -32,16 +32,10 @@ pub enum Stability {
     Deprecated(&'static str, Option<&'static str>),
 }
 
-struct Context<'a> {
-    features: &'a Features,
-    parse_sess: &'a ParseSess,
-    plugin_attributes: &'a [(Symbol, AttributeType)],
-}
-
 macro_rules! gate_feature_fn {
```

Comment:
Aside: In a follow-up, I think we should consider making this infra not use macros since it tends to be infectious atm and not very readable.